.github/actions/build-maven/action.yaml

```diff
@@ -0,0 +1,101 @@
+name: ""
+description: ""
+
+inputs:
+  module:
+    description: 'Module to build'
+    required: true
+  release:
+    description: 'Release?'
+    default: 'false'
+  release-version:
+    description: 'Release version'
+    required: false
+    default: "X.Y.Z"
+  next-version:
+    description: "Next version to use after release"
+    required: false
+    default: "X.Y.Z-SNAPSHOT"
+  java-version:
+    description: "Jave version used for building"
+    required: false
+    default: "21"
+  gpg-private-key:
+    description: "Gpg private key used for signing packages for maven central release"
+    required: false
+  gpg-passphrase:
+    description: "Gpg passphrase for private key"
+    required: false
+  sonatype-username:
+    description: "Sonatype username for maven central release"
+    required: false
+  sonatype-password:
+    description: "Sonatype password for maven central release"
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set up JDK
+      uses: actions/setup-java@6a0805fcefea3d4657a47ac4c165951e33482018 # v4.2.2
+      with:
+        java-version: ${{ inputs.java-version }}
+        distribution: "temurin"
+        cache: "maven"
+        cache-dependency-path: "./${{ inputs.module }}/pom.xml"
+        server-id: "central"
+        server-username: CENTRAL_USERNAME
+        server-password: CENTRAL_PASSWORD
+        gpg-private-key: ${{ inputs.gpg-private-key }}
+        gpg-passphrase: SIGN_KEY_PASS
+    - name: Maven build
+      if: ${{ inputs.release != 'true' }}
+      shell: bash
+      run: mvn -f ./${{ inputs.module }}/pom.xml --batch-mode clean install
+    - name: Maven release
+      if: ${{ inputs.release == 'true' }}
```

Comment:
ist es nicht das ganze release zeug würde ich einen seperate Datei auslagern

Reply:
Good idea 👍 
Also think it's much cleaner to split it up into `build` and `release`.